Add onCreateIsMonthDateSelectable factory method

milankovacevic-codaxy opened the pull request:
Use this factory mothod to enable/disable specific months, quarters or years on MonthPicker component.

mstijak reviewed (approved):
LGTM, except this one minor issue. Nice touch with bitwise operations...

milankovacevic-codaxy commented:
I've updated the factory method to pass the validation parameters as the first argument (instance is the second argument). I've also updated the method's declaration file and the Litmus demo component accordingly.
